pkg/proxy/kafka.go

```diff
@@ -356,6 +356,14 @@ func (k *kafkaRedirect) handleRequestConnection(pair *connectionPair) {
 	}), "Proxying request Kafka connection")
 
 	handleRequests(k.socket.closing, pair, pair.Rx, nil, k.handleRequest)
+
+	// The proxymap contains an entry with metadata for the receive side of the
+	// connection, remove it after the connection has been closed.
+	if pair.Rx != nil {
+		if err := k.redirect.removeProxyMapEntryOnClose(pair.Rx.conn); err != nil {
```

Comment:
It can take up to 1 minute for the rx connection to get actually closed, because that’s the delay we set for SO_LINGER. Therefore, we need to wait for at least 1 minute before removing the proxy map entry. Otherwise, the connection may still hang for the client.

Reply:
> If 1 minute is too long, we can reduce the SO_LINGER delay. But we should always wait for at least that delay before removing the map entry.

Good point. I'm not sure lingering is actually better though. 1 minute is definitely too long just to send out some existing data. I agree that we should wait though.